src/routes/deposit.js

```diff
@@ -49,12 +49,10 @@ export default class Deposit extends React.Component {
       10 ** this.selectedToken.decimals
     );
 
-    bridge
-      .deposit(this.selectedToken, value)
-      .on('transactionHash', depositTxHash => {
-        console.log('deposit', depositTxHash); // eslint-disable-line
-        this.setState({ value: 0 });
-      });
+    bridge.deposit(this.selectedToken, value).then(({ transactionHash }) => {
```

Comment:
Promise will resolve only when the transaction will be mined. So field will be cleared after several seconds, but we need to clear it after transaction sending (when we have `transactionHash`)

Reply:
Fixing!